features/junit_format.feature

```diff
@@ -42,15 +42,15 @@ Feature: JUnit Formatter
           <testcase name="Failed" classname="Adding numbers" status="failed" time="-IGNORE-VALUE-" file="features-DIRECTORY-SEPARATOR-single_feature.feature" line="25">
             <failure message="Then I must have 13: Failed asserting that 14 matches expected '13'."/>
           </testcase>
-          <testcase name="Passed &amp; Failed #1" classname="Adding numbers" status="failed" time="-IGNORE-VALUE-" file="features-DIRECTORY-SEPARATOR-single_feature.feature" line="29">
+          <testcase name="Passed &amp; Failed with value=5 #1" classname="Adding numbers" status="failed" time="-IGNORE-VALUE-" file="features-DIRECTORY-SEPARATOR-single_feature.feature" line="29">
```

Comment:
I have not explicitly added a test to prove that the `#1`, `#2` etc are in fact consistent between runs. 

However, we're proving the placeholders appear in the output, and we know that comes from the implementation in Behat/Gherkin that also appends the index as it parses the gherkin file - and that is tested in that project. I think that provides sufficient confidence that the rerun will now be numbered correclty.